### PR TITLE
Spies can be moved on map

### PR DESCRIPTION
**Pull request by tuvus**
This PR adds on to #7610 by adding a new more visual way of moving spies using the world screen.

Right clicking the move button will close the espionage screen and open the world screen with the spy selected. The player can then move the spy to any available city that the spy can move to. Available cities have a cyan circle on them _(I chose cyan because I like it and it stands out. This can definitely be changed. Maybe it should depend on the resulting action in the city)_. Clicking on the tile shows a move button that moves the spy there when clicked and opens the espionage screen.

Pressing esc or the X button on the spy panel exits this mode.

<details><summary>Details</summary>

MoveSpyViaMap

</details>

### CODE REVIEW

**Review by yairm210 — APPROVED** [review]

File: core/src/com/unciv/ui/screens/diplomacyscreen/MajorCivDiplomacyTable.kt [hunk 1]
```diff
@@ -85,12 +85,12 @@ class MajorCivDiplomacyTable(private val diplomacyScreen: DiplomacyScreen) {
         if (otherCiv.cities.isNotEmpty() && otherCiv.getCapital() != null && viewingCiv.hasExplored(otherCiv.getCapital()!!.getCenterTile()))
             diplomacyTable.add(diplomacyScreen.getGoToOnMapButton(otherCiv)).row()
 
-        if (!otherCiv.isHuman()) { // human players make their own choices
+//        if (!otherCiv.isHuman()) { // human players make their own choices
```
Comment (yairm210) [inline core/src/com/unciv/ui/screens/diplomacyscreen/MajorCivDiplomacyTable.kt:88]: Why this change?

Comment (tuvus) [reply core/src/com/unciv/ui/screens/diplomacyscreen/MajorCivDiplomacyTable.kt:88]: Oooops, that slipped through. I was testing if the modifiers worked correctly.

**Comment by tuvus** [comment]
I just added another way to close out of the spy moving on map mode. The player can now click on any ineligible tile to click on an X button which pops up, bringing them back to the espionage screen. This should make it more obvious how to cancel the spy movement.

<details><summary>Picture</summary>

MoveSpyViaMap2


</details>
